crates/da-clients/aptos/src/lib.rs

```diff
@@ -0,0 +1,248 @@
+        let data = prepare_blob(&state_diff, &self.trusted_setup).await?;
+        let loop_cycle = get_env_var_or_panic("LOOP_CYCLE").parse::<usize>()?;
+
+        // TODO: It’s better to use .unzip() for cleaner code.
+        let (blobs, commitments, proofs) = data
+            .into_iter()
+            .map(|(blob, commitment, proof)| {
+                (
+                    // Split blobs into "loop_cycle" parts.
+                    blob.to_vec()
+                        .chunks(BYTES_PER_BLOB / loop_cycle)
+                        .map(|part| vec![MoveValue::vector_u8(part.to_vec())])
+                        .collect::<Vec<_>>(),
+                    MoveValue::vector_u8(commitment.to_vec()),
+                    MoveValue::vector_u8(proof.to_vec()),
+                )
+            })
+            .fold((vec![], vec![], vec![]), |(mut blobs, mut commitments, mut proofs), (blob, commitment, proof)| {
+                blobs.push(blob);
+                commitments.push(commitment);
+                proofs.push(proof);
+                (blobs, commitments, proofs)
+            });
+
+        // Build and sign transaction
+        let sequence_number = self.client.get_account(self.account.address()).await?.into_inner().sequence_number;
+        self.account.set_sequence_number(sequence_number);
+        let mut txs = vec![];
+
+        for i in 0..loop_cycle {
+            let payload = TransactionPayload::EntryFunction(EntryFunction::new(
+                ModuleId::new(self.account.address(), Identifier::new(STARKNET_VALIDITY).unwrap()),
+                Identifier::new(BLOB_SUBMISSION).unwrap(),
+                vec![],
+                serialize_values(vec![
+                    &MoveValue::Vector(blobs.first().unwrap().get(i).unwrap().to_vec()),
+                    &MoveValue::Vector(commitments.clone()),
+                    &MoveValue::Vector(proofs.clone()),
+                ]),
+            ));
+            let tx = build_transaction(payload, &self.account, self.chain_id);
+            txs.push(tx);
+        }
+
+        // Submit transaction.
+        let pending_transactions = txs
+            .into_iter()
+            .map(|tx| {
+                let client = self.client.clone();
+                tokio::spawn(async move {
+                    
+                    client.submit(&tx).await.unwrap().into_inner()
+                })
+            })
+            .collect::<Vec<_>>();
+
+        let mut results = Vec::with_capacity(pending_transactions.len());
+        for handle in pending_transactions {
+            results.push(handle.await.unwrap());
+        }
+
+        // Wait for transaction.
+        let results = results
+            .into_iter()
+            .map(|pending_tx| {
+                let client = self.client.clone();
+                tokio::spawn(async move {
+                    let transaction = client.wait_for_transaction(&pending_tx).await.unwrap().into_inner();
```

Comment:
handle error

Reply:
@Tranduy1dol r u sure you resolved this?